torchdata/datapipes/iter/load/s3io.py

```diff
@@ -108,20 +120,32 @@ class S3FileLoaderIterDataPipe(IterDataPipe[Tuple[str, StreamWrapper]]):
         multi_part_download: flag to split each chunk into small packets and download those packets in parallel (enabled by default)
 
     Example:
-        >>> from torchdata.datapipes.iter import IterableWrapper, S3FileLoader
-        >>> dp_s3_urls = IterableWrapper(['s3://bucket-name/folder/', ...]).list_files_by_s3()
+
+    .. testsetup::
+
+        from torchdata.datapipes.iter import IterableWrapper, S3FileLoader
+
+        S3FileLoader.__iter__ = lambda self: iter([])
+
+    .. testcode::
+
+        from torchdata.datapipes.iter import IterableWrapper, S3FileLoader
```

Comment:
Imports are duplicated here, is this on purpose?

Reply:
Yes because S3FileLoader needs to be imported in the test setup to patch it. And in the example that is displayed to the user, the user should see what to import in order to run the code.